### ISSUE TITLE
Themes work only partially

### ISSUE THREAD

**Issue by ZwergNaseXXL** [MarcBoule/ImpromptuModular#77]
Hi Marc, I noticed some issues with the panel themes a while ago but never bothered to look into it. Now I did and that functionality seems to be broken in a few ways:

* "Use Rack global theme" doesn't seem to have any effect at all.
* "Use Rack global theme" and "Dark" can both be enabled at the same time, which doesn't make much sense.
* All the modules are always added with the light theme and you have to switch every single one to dark. Would be nice if the plugin remembers the last setting for newly added modules. "themeDefault" from the settings file is apparently not used anymore.
* Some modules (expanders?) are missing the theme options and can only be added in light.

It's only a minor annoyance, but for Impromptu modules I have always preferred the dark theme and it would be nice to have it back.

**Comment by MarcBoule** [comment]
Hi, can you tell me the version of Rack and the version of Impromptu you are using? I'm not noticing some of the behaviors you mention. In my tests below, I have Rack 2.5.2 and Impromptu 2.4.3.

* When an Impromptu module has its "Use Rack global theme" option turned on, which is the default, then the module's theme will follow the global theme, and in a quick test, when I toggle the global theme in Rack's View menu, I can see my modules change color accordingly, and everything looks normal.

* "Use Rack global theme" and "Dark" are not really both enabled simultaneously, since the "Dark" goes dark grey to show that the option is not in use (Rack should also gray out the checkmark also, but doesn't, which might lead to the impression that both menus are active). Here's how it should appear:

![image](https://github.com/MarcBoule/ImpromptuModular/assets/33604154/769801c0-ffb6-4bd7-bfca-ba18b49b44f6)

* Indeed, the global Impromptu setting is no longer used, but the plugin will automatically create the new modules according to the global theme selected in Rack. So if your global Rack setting is set to prefer dark panels, all intantiated Impromptu modules should be dark (and vice-versa). If you are not seeing this, something is weird.

* Expanders have no theme settings since they are meant to automatically be the same color as the main module they are connected to. Here I am seeing a bug, since I was able to reach a case where an expander is dark while its main module is not dark.

So only the 4th item you raise seems like a bug to me, but I'll need more info and details if you are seeing something inconsistent with what i wrote above.

Thanks for the feedback! Cheers,

Marc

**Comment by ZwergNaseXXL** [comment]
Oh right, I didn't think about the versions because usually I am up-to-date, but I hesitated updating to Rack 2.5 because of the changed user folder location. So I have Rack 2.4.1 and Impromptu 2.4.3. I'm not aware of any changes in Rack's theme API though.

Anyway, I just remembered that there are two theme-related settings in Rack's "View" menu: "Theme" and "Use dark panels if available (experimental)". I usually have the first one at dark and the second at off. When I reported this issue, I only experimented with the "Theme" setting and it did have no effect. Instead, Impromptu modules follow the "Use dark panels" option.

So I guess everything works like you intended. But it means for me that I would either have to enable dark panels globally (which I don't want) or that I have to embrace the light theme for Impromptu, because changing it for every module instance is too annoying. Unless you'd consider adding a default theme other than Rack's or just remembering the last setting.

Oh well. Sorry for bothering you with a non-issue, but at least you found a little bug through it.
